File: ContainsDuplicates/containsDuplicatesv1.java
package ContainsDuplicates;

import java.util.HashMap;

public class containsDuplicatesv1 {
    public static void main(String[] args) {
        System.out.println(ContainsDuplicates(new int[]{1,2,3,1}));
        System.out.println(ContainsDuplicates(new int[]{1,2,3,4}));
        System.out.println(ContainsDuplicates(new int[]{1,1,1,3,3,4,3,2,4,2}));
    }

    public static boolean ContainsDuplicates(int[] nums){
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int num : nums) {
            if (map.containsKey(num)) {map.put(num, map.get(num) + 1);}
            else {map.put(num, 1);}
        }

        for (int key : map.keySet()) {
            if (map.get(key) > 1) {
                return true;
            }
        }
        return false;
    }
}

File: 217. ContainsDuplicates/ContainsDuplicatesv1.py
def ContainsDuplicates(nums):
    """
    Given an integer array nums, return true if any value appears at least twice in the array,
    and return false if every element is distinct.
    """
    num_set = set()
    for num in nums:
        if num in num_set:
            return True
        num_set.add(num)
    return False

if __name__ == "__main__":
    # Example usage
    nums = [1, 2, 3, 1]
    print(ContainsDuplicates(nums))  # Output: True
